hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataXceiver.java

```diff
@@ -632,6 +632,11 @@ public void readBlock(final ExtendedBlock block,
       datanode.metrics.incrBytesRead((int) read);
       datanode.metrics.incrBlocksRead();
       datanode.metrics.incrTotalReadTime(duration);
+      if (read < 0 || duration <= 0) {
```

Comment:
You duplicated this logic, I would put behind a private helper so people don't forget this check.

Reply:
moved to a Utils method